File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        m={}
        maxi=None
        for num in nums:
            if num not in m:
                m[num]=1
            else:
                m[num]+=1
        x=sorted(m.items(),key=lambda x:x[1],reverse=True)
        return x[0][0]            

File: 0169-majority-element/0169-majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let num=null;
    let counter=0
    for (x of nums){
        if (counter===0){
            num=x;
            counter++
        }else{
            if (num===x){
                counter++;
            }else{
                counter--;
            }
        }
    }
    return num;
    
};